x-pack/metricbeat/module/gcp/metrics/metrics_requester.go

```diff
@@ -252,6 +252,10 @@ func (r *metricsRequester) getFilterForMetric(serviceName, m string) string {
 	// to apply any additional filters.
 	if locationsConfigsAvailable && !isAGlobalService(serviceName) {
 		serviceLabel := getServiceLabelFor(serviceName)
+		// If user has entered a location label, overide the serviceLabel with the locationlabel
+		if r.config.LocationLabel != "" {
+			serviceLabel = r.config.LocationLabel
+		}
```

Comment:
nit:

```diff
diff --git a/x-pack/metricbeat/module/gcp/metrics/metrics_requester.go b/x-pack/metricbeat/module/gcp/metrics/metrics_requester.go
index 66f08dbab5..55490da875 100644
--- a/x-pack/metricbeat/module/gcp/metrics/metrics_requester.go
+++ b/x-pack/metricbeat/module/gcp/metrics/metrics_requester.go
@@ -251,11 +251,7 @@ func (r *metricsRequester) getFilterForMetric(serviceName, m string) string {
 	// NOTE: some GCP services are global, not regional or zonal. To these services we don't need
 	// to apply any additional filters.
 	if locationsConfigsAvailable && !isAGlobalService(serviceName) {
-		serviceLabel := getServiceLabelFor(serviceName)
-		// If user has entered a location label, overide the serviceLabel with the locationlabel
-		if r.config.LocationLabel != "" {
-			serviceLabel = r.config.LocationLabel
-		}
+		serviceLabel := r.determineServiceLabel(serviceName)
 		f = r.buildLocationFilter(serviceLabel, f)
 	}
 
@@ -265,6 +261,16 @@ func (r *metricsRequester) getFilterForMetric(serviceName, m string) string {
 	return f
 }
 
+// determineServiceLabel determines the service label to use for the given service name. If a custom
+// location label is configured, it will be used. Otherwise, the default service label for the
+// given service name will be returned.
+func (r *metricsRequester) determineServiceLabel(serviceName string) string {
+	if r.config.LocationLabel != "" {
+		return r.config.LocationLabel
+	}
+	return getServiceLabelFor(serviceName)
+}
+
 // Returns a GCP TimeInterval based on the ingestDelay and samplePeriod from ListMetricDescriptor
 func getTimeIntervalAligner(ingestDelay time.Duration, samplePeriod time.Duration, collectionPeriod *durationpb.Duration, inputAligner string) (*monitoringpb.TimeInterval, string) {
 	var startTime, endTime, currentTime time.Time
```

Just to make it cleaner. Rest of the changes looks good!